File: RailgunVB/Core/Configuration/DatabaseConfig.vb
Imports Newtonsoft.Json

Namespace Core.Configuration
    
    <JsonObject(MemberSerialization.Fields)>
    Public Class DatabaseConfig
    
        Public ReadOnly Property Hostname As String
        Public ReadOnly Property Username As String
        Public ReadOnly Property Password As String
        Public ReadOnly Property Database As String

        <JsonConstructor>
        Public Sub New(hostname As String, username As String, password As String, database As String)
            Me.Hostname = hostname
            Me.Username = username
            Me.Password = password
            Me.Database = database
        End Sub
        
    End Class
    
End NameSpace

File: Railgun/Core/Configuration/DatabaseConfig.cs
using Newtonsoft.Json;

namespace Railgun.Core.Configuration
{
    [JsonObject(MemberSerialization.Fields)]
    public class DatabaseConfig
    {
        public string Hostname { get; }
        public string Username { get; }
        public string Password { get; }
        public string Database { get; }

        [JsonConstructor]
        public DatabaseConfig(string hostname, string username, string password, string database) {
            Hostname = hostname;
            Username = username;
            Password = password;
            Database = database;
        }
    }
}